### DISCUSSION TITLE
Switching finer granular metrics/logs/traces

### DISCUSSION THREAD

**Discussion by prathyk** [open-telemetry/opentelemetry-specification#1567] [Q&A]
### Enabling and Disabling logger levels and possibly the level of metrics and traces
log4j or other logging systems have a way to name the loggers. And have multiple levels of logging (DEBUG, INFO, TRACE etc.)
otel has a way to name metrics and [traces](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#get-a-tracer). But it has no concept of levels. 
There is also no standard language agnostic way to enable/disable specific loggers or specific named tracers dynamically on a running system. 

In other words, 
_Is there a way by which I can enable specific logger dynamically on a app which is instrumented using otel. Why does otel not specify such a standard?_

#### Factors to consider
1. The api to enable a specific logger/tracer should consider all scopes (pods, deployment etc.). i.e enabling a logger for a service should enable it for all pods in that service.
2. I should be able to enable tracing only for a specific request. 

Any thoughts?

**Comment by kumoroku** [comment]
regarding log levels, have a look at the spec here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#severity-fields (the spec is using the term "Severity")

**Reply by prathyk** [reply]
okay thanks. 
The spec should standardize the apis to change severity of logs/traces also.

**Comment by prathyk** [comment]
I'm not sure why this question is not receving much traction. I'm still looking for an answer. 
Do you guys think that this topic is irrelevant to opentelemetry? Do we have other ways to accomplish this?

**Reply by jkwatson** [reply]
I suspect it's not receiving much traction because others don't have the same concerns or needs that you do. If you think this is an important feature for OpenTelemetry, I recommend engaging with the specification SIG and submitting an OTEP for your feature.

**Reply by yurishkuro** [reply]
I think there's no traction because there's no proposal in the OP. What OP essentially says is "I have this problem, someone please solve it for me", which means someone else needs to drop what else they are doing and do the research on how to solve this problem. But other people may have other priorities. There are a lot of open questions when it comes to tracing, we can't be solving them all at once.
